nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -0,0 +1,339 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.mastodon;
+
+  env = {
+    RAILS_ENV = "production";
+    NODE_ENV = "production";
+
+    VAPID_PUBLIC_KEY = cfg.vapidPublicKey;
+    DB_USER = cfg.dbUser;
+    SMTP_LOGIN  = cfg.smtpLogin;
+
+    REDIS_HOST = cfg.redisHost;
+    REDIS_PORT = toString(cfg.redisPort);
+    DB_HOST = cfg.dbHost;
+    DB_PORT = toString(cfg.dbPort);
+    DB_NAME = cfg.dbName;
+    LOCAL_DOMAIN = cfg.localDomain;
+    SMTP_SERVER = cfg.smtpServer;
+    SMTP_PORT = toString(cfg.smtpPort);
+    SMTP_FROM_ADDRESS = cfg.smtpFromAddress;
+    PAPERCLIP_ROOT_PATH = "/var/lib/mastodon/public-system";
+    PAPERCLIP_ROOT_URL = "/system";
+    ES_ENABLED = if (cfg.elasticsearchHost != null) then "true" else "false";
+    ES_HOST = cfg.elasticsearchHost;
+    ES_PORT = toString(cfg.elasticsearchPort);
+  };
+
+in {
+
+  options = {
+    services.mastodon = {
+      enable = lib.mkEnableOption "Mastodon federated social network";
+      configureNginx = lib.mkEnableOption ''
```

Comment:
This should be `lib.mkOption`, because the argument to `mkEnableOption` is the name, meaning the description of this option is going to be

```
Whether to enable Configure nginx as reverse proxy for mastodon
Alternatively you can configure a reverse-proxy of your choice to serve these paths:
…
https://github.com/tootsuite/mastodon/blob/master/dist/nginx.conf.
```

Reply:
I'm a bit confused, where should this text go instead? There is only one description/name for one option or is there a separate attribute for an extended description?
Until now it was intended to describe the alternative in the description of configureNginx, since I didn't know where to put it.

Reply:
The problem with `mkEnableOption` is it tries to synthesize a description from a name, but when you're providing a full description, that doesn't particularly work. I'm suggesting

```nix
configureNginx = lib.mkOption {
  default = false;
  example = true;
  description = ''
    Configure nginx as reverse proxy for mastodon
    Alternatively you can configure a reverse-proxy of your choice to serve these paths:
    …
    https://github.com/tootsuite/mastodon/blob/master/dist/nginx.conf
  '';
  type = lib.types.bool;
};
```